### PR TITLE
More datastructure lemmas.

### PR DESCRIPTION
**Pull request by emina**
This PR removes deprecated List lemmas and adds more useful lemmas about Lists and Sets.

### CODE REVIEW

File: cedar-lean/Cedar/Thm/Data/Set.lean [hunk 1]
```diff
@@ -321,6 +353,74 @@ theorem inter_wf {α} [LT α] [StrictLT α] [DecidableLT α] [DecidableEq α] {s
...
+  simp only [intersects, List.any_eq_true, in_list_iff_in_set, contains_prop_bool_equiv]
+  constructor
+  case mp =>
+    intro h
+    replace ⟨x, h⟩ := h
+    by_contra hc
+    simp only [ne_eq, Bool.not_eq_false] at hc
+    simp only [empty_iff_not_exists, not_exists] at hc
+    specialize hc x
+    have _ := mem_inter_iff.mpr h
```
Comment (cdisselkoen) [inline cedar-lean/Cedar/Thm/Data/Set.lean:400]:
```suggestion
    have := mem_inter_iff.mpr h
```

File: cedar-lean/Cedar/Thm/Data/Set.lean [hunk 2]
```diff
@@ -38,6 +38,22 @@ theorem if_wellformed_then_exists_make [LT α] [DecidableLT α] (s : Set α) :
...
 
+theorem wf_iff_sorted {α} [LT α] [DecidableLT α] [StrictLT α]  (s : Set α) :
+  s.WellFormed ↔ s.elts.Sorted
+:= by
+  simp only [WellFormed, make, toList, elts]
+  have hid : (fun x => x) = @id α := by
+    apply funext ; simp only [id_eq, implies_true]
+  rw [hid]
+  constructor <;>
+  intro h
```
Comment (cdisselkoen) [inline cedar-lean/Cedar/Thm/Data/Set.lean:49]: In other places (like line 926 above) you have this on one line.  Which is preferable?

File: cedar-lean/Cedar/Thm/Data/Set.lean [hunk 3]
```diff
@@ -179,6 +199,14 @@ theorem make_wf [LT α] [DecidableLT α] [StrictLT α] (xs : List α) :
 := by
   simp only [WellFormed, make, toList, elts, List.canonicalize_idempotent]
 
+theorem make_sorted {α} [LT α] [DecidableLT α] [StrictLT α] {xs : List α} :
+  xs.Sorted → Set.make xs = Set.mk xs
+:= by
+  intro h
+  have hid : (fun x => x) = @id α := by
+    apply funext ; simp only [id_eq, implies_true]
```
Comment (cdisselkoen) [inline cedar-lean/Cedar/Thm/Data/Set.lean:207]: Is this not true by definitional equality?  Does the `simp only` below fail if you don't include `hid` as a lemma?

**Review by cdisselkoen — APPROVED** [review]
Looks good

File: cedar-lean/Cedar/Thm/Data/Set.lean (hunk 1 above)
Comment (emina) [reply cedar-lean/Cedar/Thm/Data/Set.lean:400]: Ah, I use `have _ := ` to make it clear that the introduced hypothesis will never be used by name.  Note that `have := ` introduces a usable name into the proof context (`this`).

File: cedar-lean/Cedar/Thm/Data/Set.lean (hunk 2 above)
Comment (emina) [reply cedar-lean/Cedar/Thm/Data/Set.lean:49]: No thought out preference though empirically, it appears that I favor putting things on line :D

File: cedar-lean/Cedar/Thm/Data/Set.lean (hunk 1 above)
Comment (cdisselkoen) [reply cedar-lean/Cedar/Thm/Data/Set.lean:400]: I see

File: cedar-lean/Cedar/Thm/Data/Set.lean (hunk 3 above)
Comment (emina) [reply cedar-lean/Cedar/Thm/Data/Set.lean:207]: Surprisingly, this is necessary, and `simp only` does indeed fail without it.

Comment (cdisselkoen) [reply cedar-lean/Cedar/Thm/Data/Set.lean:207]: Weird

**Review by khieta — APPROVED** [review]

**Pull request merged**